### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by George3d6**
Almost a full refacotr of the python SDK.

This will also include a pandas connector.

Not yet complete since the testing is not yet complete and I'm finding a lot of bugs in the API while testing.

### CODE REVIEW

File: tests/test_predictors.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_2_train(self):
+        try:
+            del self.predictors['test_predictors_1']
+        except Exception as e:
+            print(e)
+        self.predictors.learn('test_predictors_1', 'test_2_file_datasource', 'y', args={
+            'stop_training_in_x_seconds': 30
+        })
+        time.sleep(3)
```
Comment (StpMax) [inline tests/test_predictors.py:32]: often 3sec not enough for me

File: tests/test_predictors.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        pred = self.predictors['test_predictors_1']
+        self.assertTrue('status' in pred.get_info())
```
Comment (StpMax) [inline tests/test_predictors.py:34]: `pred.get_info()` raise error if predictor not exists. May be return `None` in this case?

File: mindsdb_sdk/__about__.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+__title__ = 'mindsdb_sdk'
+__package_name__ = 'mindsdb_sdk'
+__version__ = '0.10.0'
+__description__ = "MindsDB Python SDK, provides an SDK to use a remote mindsdb instance"
+__email__ = "jorge@mindsdb.com"
+__author__ = 'MindsDB Inc'
+__github__ = 'https://github.com/mindsdb/mindsdb_python_sdk'
+__pypi__ = 'https://pypi.org/project/mindsdb-sdk/'
+__license__ = 'MIT'
```
Comment (StpMax) [inline mindsdb_sdk/__about__.py:9]: GNU

File: mindsdb_sdk/__about__pd.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+__title__ = 'pd_automl'
+__package_name__ = 'pd_automl'
+__version__ = '0.1.0'
+__description__ = "Pd automl provides a pandas dataframe accessor to allow using dataframes with MindsDB (either the local native library or a local or remote mindsdb server)"
+__email__ = "jorge@mindsdb.com"
+__author__ = 'MindsDB Inc'
+__github__ = 'https://github.com/mindsdb/mindsdb_python_sdk'
+__pypi__ = 'https://pypi.org/project/pd_automl/'
+__license__ = 'MIT'
```
Comment (StpMax) [inline mindsdb_sdk/__about__pd.py:9]: GNU

File: mindsdb_sdk/__init__.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import sys
+if sys.version_info < (3,5):
+    raise EnvironmentError('Please install a python version >= 3.5 to use this library')
```
Comment (StpMax) [inline mindsdb_sdk/__init__.py:3]: 3.6?

File: tests/test_predictors.py (hunk 1 above)
Comment (StpMax) [inline tests/test_predictors.py:32]: often 3sec not enough for me

File: tests/test_predictors.py (hunk 2 above)
Comment (StpMax) [inline tests/test_predictors.py:34]: `pred.get_info()` raise error if predictor not exists. May be return `None` in this case?

File: tests/test_predictors.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_3_predict(self):
+        pred = self.predictors['test_predictors_1']
+        while pred.get_info()['status'] != 'complete':
+            print('Predictor not done trainig, status: ', pred.get_info()['status'])
+            time.sleep(3)
+
+        prediction = pred.predict(datasource={'theta3': 1})
+        self.assertTrue('prediction' in prediction)
+        self.assertTrue('prediction' in prediction)
```
Comment (StpMax) [inline tests/test_predictors.py:44]: tests fail on this assert. For me at this point
```
prediction = [{'y': {'predicted_value': 0.01983735508131215, 'confidence': 0.9725, 'prediction_quality': 'very confident', 'confidence_interval': [-0.24289705720733626, 0.28257176736996054], 'important_missing_information': ['theta4', 'theta5'], 'confidence_composition': {}, 'extra_insights': {'if_missing': [{'theta3': 0.12333724603272951}]}}}]
```
so is no 'prediction' inside.
also last line is duplicated

**Review by StpMax — CHANGES_REQUESTED** [review]

File: tests/test_predictors.py (hunk 1 above)
Comment (George3d6) [reply tests/test_predictors.py:32]: Increased, but why is this happening in the first place ? Shouldn't the predictor be query-able right away ?

File: tests/test_predictors.py (hunk 2 above)
Comment (George3d6) [reply tests/test_predictors.py:34]: disagree, but we should address this via error on init if predictor doesn't exist

File: mindsdb_sdk/__about__.py (hunk 3 above)
Comment (George3d6) [reply mindsdb_sdk/__about__.py:9]: GPL-3.0 I believe, but yeah, very good point !

File: mindsdb_sdk/__about__pd.py (hunk 4 above)
Comment (George3d6) [reply mindsdb_sdk/__about__pd.py:9]: GPL-3.0 I believe, but yeah, very good point !

**Pull request merged**